tests/data/test_datamodule.py

```diff
@@ -61,3 +64,45 @@ def test_dm_from_dataset(setup_fn: Callable, collate_fn: Callable, collate_fn_ar
 
     # iterate through the dataloaders
     iterate_dataloaders(datamodule)
+@pytest.mark.cpu
+@pytest.mark.parametrize("limit_len", [2, (2, 3, 1), (10, 10, 10)])
+@pytest.mark.parametrize("ds_type", ["hf", "torch"])
+def test_limit_datasets(ds_type: str, limit_len: int | tuple[int, int, int]) -> None:
+    """Test the `limit_datasets` method of PrunaDataModule with various dataset types and limits."""
+
+    # Create small datasets
+    if ds_type == "hf":
+        data = {"text": ["a", "b", "c", "d", "e"]}
+        train_ds = val_ds = test_ds = Dataset.from_dict(data)
+    elif ds_type == "torch":
+        x = torch.arange(5).float().unsqueeze(1)
+        y = torch.arange(5)
+        train_ds = val_ds = test_ds = TensorDataset(x, y)
+    else:
```

Comment:
I guess we can remove this `else` as we specify the `ds_type` :thinking: , wdyt?